game.js

```diff
@@ -0,0 +1,217 @@
+                + numbersAroundButtonRight[k]]);
+                btnDeleted.style.opacity = '0';
+              } else if (buttonsForDelete[i] === showEmptyFields.upperEdges[j]) {
+                const btnDeleted = document.getElementById(board.buttonId[btnDeletedInt
+                + numbersAroundButtonUp[k]]);
+                btnDeleted.style.opacity = '0';
+              } else if (buttonsForDelete[i] === board.buttonId[0]) {
+                if (k < leftUpNumbers.length) { // k will be over arrays length
+                  const btnDeleted = document.getElementById(board.buttonId[btnDeletedInt
+                  + leftUpNumbers[k]]);
+                  btnDeleted.style.opacity = '0';
+                }
+              } else if (buttonsForDelete[i] === board.buttonId[board.x - 1]) {
+                if (k < rightUpNumbers.length) {
+                  const btnDeleted = document.getElementById(board.buttonId[btnDeletedInt
+                  + rightUpNumbers[k]]);
+                  btnDeleted.style.opacity = '0';
+                }
+              } else if (buttonsForDelete[i] === board.buttonId[board.size - board.x]) {
+                if (k < leftDownNumbers.length) {
+                  const btnDeleted = document.getElementById(board.buttonId[btnDeletedInt
+                  + leftDownNumbers[k]]);
+                  btnDeleted.style.opacity = '0';
+                }
+              } else if (buttonsForDelete[i] === board.buttonId[board.size - 1]) {
+                if (k < rightDownNumbers.length) {
+                  const btnDeleted = document.getElementById(board.buttonId[btnDeletedInt
+                  + rightDownNumbers[k]]);
+                  btnDeleted.style.opacity = '0';
+                }
+              }
+            }
+          }
+        }
+        for (let i = 0; i < buttonsForDelete.length; i++) {
+          for (let j = 0; j < buttonIdForDelete.length; j++) {
+            for (let k = 0; k < numbersAroundButton.length; k++) {
+              if (buttonsForDelete[i] === buttonIdForDelete[j]) {
+                const btnDeletedInt = Number(buttonsForDelete[i]);
+                const btnDeleted = document.getElementById(board.buttonId[btnDeletedInt
+                + numbersAroundButton[k]]);
+                btnDeleted.style.opacity = '0';
+              }
+            }
+          }
+        }
+      }
+      if (btnWithNum !== 1) {
+        buttonDelete();
+        for (let i = 0; i < board.buttonId.length; i++) {
+          const getButton = document.getElementById(board.buttonId[i]);
+          if (getButton.style.opacity === '0') {
+            if (getButton.style.background !== 'grey') { // if user left a flag and button disappeared cause there was no bomb the mines counter sums the count of disappeared flags
+              minesLeftCounter++;
+              minesCounterHtml.value = minesLeftCounter;
+            }
+          }
+        }
+      }
+    });
```

Comment:
this code once again is overcomplicated. Most of the problems are connected with wrong data structures. Please redesign it according to mu suggestion. Those kind of code is not acceptable please keep it simple.